tests/cases/fourslash/completionsConditionalMember.ts

```diff
@@ -0,0 +1,22 @@
+/// <reference path="fourslash.ts" />
+
+////declare function f<T extends string>(
+////  p: { a: T extends 'foo' ? { x: string } : { y: string } }
+////): void;
+////
+////f<'foo'>({ a: { /*1*/ } });
+////f<string>({ a: { /*2*/ } });
```

Comment:
Why no tests without type arguments? The new code seems to apply exclusively in this case.

Reply:
[completionsGenericIndexedAccess2](https://github.com/microsoft/TypeScript/pull/34855/files#diff-10da2e23de6b7208e06ed04e2e1770c5) is without type arguments, which tests the primary way that #33937 was a regression. Almost all the existing tests from #33937 / #32100 are without type arguments, but they weren’t complex enough to fail. This test was to ensure that the new clever stuff _doesn’t_ apply when there _is_ an explicit type argument.